src/pages/GetStarted/CustomWorkspaceTab/DevfileSelector/index.tsx

```diff
@@ -79,9 +81,15 @@ export class DevfileSelectorFormGroup extends React.PureComponent<Props, State>
   private async handleDevfileSelect(meta: che.DevfileMetaData): Promise<void> {
     // clear location input
     this.devfileLocationRef.current?.clearInput();
+    const cheDevworkspaceEnabled = this.props.workspacesSettings['che.devworkspaces.enabled'] === 'true';
+    let devfile: api.che.workspace.devfile.Devfile;
     try {
-      const devfileContent = await this.props.requestDevfile(meta.links.self) as string;
-      const devfile = safeLoad(devfileContent);
+      if (cheDevworkspaceEnabled) {
```

Comment:
I think if statement can be moved out of try

Reply:
I think it is not a good idea. Next line is a request:
```
await this.props.requestFactoryResolver(meta.links.v2);
```

Reply:
We have a request which depends on boolean variable(one request or another).

Reply:
you're right. My fault